cdap-ui/app/features/tracker/controllers/integrations-ctrl.js

```diff
@@ -0,0 +1,244 @@
+
+    this.navigatorInfo = {
+      brokerString: '',
+      hostname: '',
+      username: '',
+      password: ''
+    };
+
+    this.chartSettings = {
+      chartMetadata: {
+        showx: false,
+        showy: false,
+        legend: {
+          show: false,
+          position: 'inset'
+        }
+      },
+      color: {
+        pattern: ['#35c853'] // @tracker-green
+      },
+      isLive: true,
+      interval: 1000,
+      aggregate: 5
+    };
+
+    this.navigatorState = {};
+
+    this.getNavigatorApp();
+
+    this.pollId = null;
+
+  }
+
+  showConfigPopup(event) {
+    event.stopPropagation();
+    this.showConfig = true;
+
+    let hideConfigPopup = () => {
+      this.showConfig = false;
+      document.removeEventListener('click', hideConfigPopup);
+    };
+    document.addEventListener('click', hideConfigPopup);
+  }
+
+  getKafkaBrokerList() {
+    this.myTrackerApi.getCDAPConfig({ scope: this.$scope })
+      .$promise
+      .then( (res) => {
+        let filtered = res.filter( (config) => {
+          return config.name === 'metadata.updates.kafka.broker.list';
+        });
+        this.navigatorInfo.brokerString = filtered[0].value;
+      });
+  }
+
+  getNavigatorApp() {
+    let params = {
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    };
+    this.myTrackerApi.getNavigatorApp(params)
+      .$promise
+      .then((res) => {
+        this.navigatorSetup.isSetup = true;
+        let config = {};
+        try {
+          config = JSON.parse(res.configuration);
+
+          this.navigatorInfo.hostname = config.navigatorConfig.navigatorHostName;
+          this.navigatorInfo.username = config.navigatorConfig.username;
+          this.navigatorInfo.password = config.navigatorConfig.password;
+          this.navigatorInfo.brokerString = config.metadataKafkaConfig.brokerString;
+
+          this.getNavigatorStatus();
+        } catch (e) {
+          console.log('Cannot parse configuration JSON');
+        }
+      }, () => {
+        this.getKafkaBrokerList();
+      });
+  }
+
+  getNavigatorStatus() {
+    let params = {
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    };
+
+    this.myTrackerApi.getNavigatorStatus(params)
+      .$promise
+      .then((res) => {
+        if (res.length) {
+          this.navigatorState = res[0];
+
+          if (res[0].status === 'RUNNING') {
+            this.navigatorSetup.isEnabled = true;
+            this.fetchMetrics();
+          } else {
+            this.dataSrc.stopPoll(this.pollId.__pollId__);
+          }
+        } else {
+          this.navigatorState = {
+            status: 'DISABLED'
+          };
+        }
+      });
+  }
+
+  toggleNavigator() {
+    let params = {
+      namespace: this.$state.params.namespace,
+      action: this.navigatorSetup.isEnabled ? 'start' : 'stop',
+      scope: this.$scope
+    };
+
+    this.myTrackerApi.toggleNavigator(params, {})
+      .$promise
+      .then( () => {
```

Comment:
Since its a single line we could probably have it as,

```
this.myTrackerApi.toggleNavigator(params, {})
      .$promise
      .then( 
        this.getNavigatorStatus.bind(this),
        this.myAlertOnValium.show.bind(null, {type: danger, content: err.data|})
      );
```

May be?